### PR TITLE
feat(restore-wallet): add qr scanner to import seed phrase

### PR DESCRIPTION
**Pull request by ZGiovas**

### CODE REVIEW

File: src/popup/components/Modals/AccountImport.vue [hunk 1]
```diff
@@ -102,10 +116,21 @@ export default defineComponent({
       router.push(store.state.loginTargetLocation);
     }
 
+    async function openScanQrModal() {
+      const scanResult = await openModal(MODAL_READ_QR_CODE, {
+        title: t('pages.send.scanAddress'),
+        icon: 'critical',
+      });
+
+      if (!scanResult) return;
```
Comment (pstachel) [inline src/popup/components/Modals/AccountImport.vue:125]:
```suggestion
      if (!scanResult) {
        return;
      }
```

**Comment by Liubov-crypto** [comment]
For me it's working with https://iancoleman.io/bip39/

File: src/popup/components/Modals/AccountImport.vue [hunk 2]
```diff
@@ -49,9 +59,11 @@ import { TranslateResult, useI18n } from 'vue-i18n';
 import { validateMnemonic } from '@aeternity/bip39';
 import { useStore } from 'vuex';
 import { useRouter } from 'vue-router';
+import { useModals } from '@/composables';
 import type { RejectCallback, ResolveCallback } from '../../../types';
-import { validateSeedLength } from '../../utils';
+import { validateSeedLength, MODAL_READ_QR_CODE } from '../../utils';
 
+import QrScanIcon from '../../../icons/qr-scan.svg?vue-component';
```
Comment (peronczyk) [inline src/popup/components/Modals/AccountImport.vue:66]: Please put the icon imports at the end of imports list.

File: src/popup/components/Modals/AccountImport.vue [hunk 3]
```diff
@@ -102,10 +116,23 @@ export default defineComponent({
...
+    async function openScanQrModal() {
+      const scanResult = await openModal(MODAL_READ_QR_CODE, {
+        title: t('pages.index.scanSeedPhrase'),
+        icon: 'critical',
+      });
+
+      if (!scanResult) {
+        return;
+      }
+      mnemonic.value = scanResult;
```
Comment (peronczyk) [inline src/popup/components/Modals/AccountImport.vue:128]:
```suggestion
      if (scanResult) {
        mnemonic.value = scanResult;
      }
```
Early return in this case makes no sense.

**Review by peronczyk — APPROVED** [review]

**Review by martinkaintas — APPROVED** [review]
🤝

**Pull request merged**